<launch>

<arg name="retracking" default="False" />

<group unless="$(arg retracking)">
  <!-- Edit the flags in this file to True for each part of the pipeline you
    want to troubleshoot. !-->
  <rosparam command="load" file="debug_flags.yaml"/>

  <include file="$(find multi_tracker)/launch/load_tracking_parameter_yamls.launch" />
</group>


<param name="multi_tracker/tracker/save_data" type="bool" value="false" />
<param name="multi_tracker/delta_video/save_data" type="bool" value="false" />


# set up src and data directories and define prefix name
<node name="set_exp_basename" pkg="multi_tracker" type="set_basename.py" output="screen" />

# launch tracking and data association nodes
<node name="tracker" pkg="multi_tracker" type="tracker_simplebuffer.py" output="screen" />
<node name="data_association" pkg="multi_tracker" type="data_association.py" output="screen" />

<node name="liveview" pkg="multi_tracker" type="liveviewer.py" output="screen" />

<node name="delta_video" pkg="multi_tracker" type="delta_video_simplebuffer.py" output="screen" />

<node name="video_player" pkg="image_view" type="image_view">
  <remap from="image" to="camera/image_raw" />
</node>

<node name="thresholded_viewer" pkg="image_view" type="image_view">
  <remap from="image" to="multi_tracker/1_thresholded" />
</node>
<node name="denoised_viewer" pkg="image_view" type="image_view">
  <remap from="image" to="multi_tracker/2_denoised" />
</node>
<node name="dilated_viewer" pkg="image_view" type="image_view">
  <remap from="image" to="multi_tracker/3_dilated" />
</node>
<node name="eroded_viewer" pkg="image_view" type="image_view">
  <remap from="image" to="multi_tracker/4_eroded" />
</node>
<node name="processed_viewer" pkg="image_view" type="image_view">
  <remap from="image" to="multi_tracker/processed_image" />
</node>

<!-- why did i move away from this again? TODO include processed image in it? -->
<!--
<node name="conditional_debug_viewer" pkg="multi_tracker" type="conditional_debug_viewer.py" output="screen" />
-->

</launch>
